docs/getting-started/openshift/index.md

```diff
@@ -0,0 +1,187 @@
+# Getting Started - OpenShift Routing
+
+This guide covers how Argo Rollouts integrates with
+[OpenShift Routing](https://docs.openshift.com/enterprise/3.0/architecture/core_concepts/routes.html) 
+for traffic management. This guide builds upon the concepts of the [basic getting started guide](../../getting-started.md).
+
+## Requirements
+- OpenShift cluster with support for routes
+- Argo Rollouts installed and running ([installation](../installation.md))
+
+!!! tip
+    Make sure you are either switched to the correct namespace or you specify the namespace to be used in the following examples. It is assumed you are working on the `argo-rollouts` namespace from the installation guide.
+
+## 1. Create Kubernetes Services
+
+The following will create two service resources named `stable-service` and `canary-service`:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: deployment-example
+  name: stable-service
+  namespace: argo-rollouts
+spec:
+  type: ClusterIP
+  ports:
+  - name: 8080-tcp
+    port: 8080
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: deployment-example
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: deployment-example
+  name: canary-service
+spec:
+  type: ClusterIP
+  ports:
+  - name: 8080-tcp
+    port: 8080
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: deployment-example 
+```
+```shell
+oc apply -f https://raw.githubusercontent.com/argoproj/argo-rollouts/master/docs/getting-started/openshift/services.yaml
+```
+
+## 2. Create OpenShift Route
+
+Easiest way to create a new route is to expose the `stable-service` through the OpenShift CLI tool:
+```shell
+oc expose service stable-service
+```
+
+This will create a new route named `stable-service` that will expose the service on a public URL.
+
+Alternatively, create and apply a route of your own:
+```yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  labels:
+    app: deployment-example
+  name: stable-service
+  namespace: argo-rollouts
+spec:
+  port:
+    targetPort: 8080-tcp
+  to:
+    kind: Service
+    name: stable-service
+    weight: 100
+```
+```shell
+oc apply -f https://raw.githubusercontent.com/argoproj/argo-rollouts/master/docs/getting-started/openshift/route.yaml
+```
+
+The URL can be found through the CLI under `Requested Host`:
+```shell
+oc describe route stable-service
+```
+ 
+## 3. Create Argo Rollout
+
+You must define at least one route under the routes field:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: openshift-rollout-demo
+  namespace: argo-rollouts
+spec:
+  replicas: 5
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      app: deployment-example
+  template:
+    metadata:
+      labels:
+        app: deployment-example
+    spec:
+      containers:
+        - image: openshift/deployment-example:v1
```

Comment:
Shall we use the "rollout-demo" to make it consistent with other examples?

Reply:
sure thing